woocommerce/woocommerce-android
# QR code login feature

This PR implementers the embedded QR code scanner code. It uses [Android ML learning kit](https://developers.google.com/ml-kit/vision/barcode-scanning/android) for it. The approach followed is using [Google Play Service](https://developers.google.com/ml-kit/vision/barcode-scanning/android#:~:text=If%20you%20choose%20to%20use%20the%20model%20in%20Google%20Play%20Services)s and downloading the ML kit on demand after app installation to avoid increasing .aab size

Please **DON'T PANIC**, I know the PR is huge, as this project required to add this embedded QR code in a different module. I picked up where @0nko left it from a hack week project. Thus, it was hard to split the PR into smaller ones. I've removed as much code as possible and only left what was needed. "Luckily" most of the file changes are small changes related to adding a new module to the project, or new resources and simple layouts. 

### Testing instructions
To test: 

- First request a magic link/QR code email
- Click on "Continue with WordPress.com" 
- Click on "Scan QR code to login" -> Check `login_with_qr_code_button_tapped, Properties: {"flow":"wordpress_com","is_debug":true}` is tracked. 
- Check that permission is requested (reject permission and retry again, grating permission) 
- Click on "Help" button inside QR code scanner screen and check it opens help screen. 
- Navigate back and scan the QR code. Verify the app logs in correctly. 
- Verify that when the code is scanned the following event is tracked: `login_with_qr_code_scanned, Properties: {"is_debug":true}`

See the screen recording for the expected behaviour: 

https://user-images.githubusercontent.com/2663464/194370715-7f5102a9-2a7f-4d81-9535-5bec572d1875.mp4
